File: src/elephant_ffi.erl
-module(elephant_ffi).
-compile([no_auto_import]).

-export([persist/2]).

persist(Name, Gen) ->
	case catch persistent_term:get(Name) of
		{ 'EXIT', { badarg, _ } } -> 
			Value = Gen(),
			persistent_term:put(Name, { ok, Value }),
			Value;
		{ ok, Value } -> Value
	end.

File: src/elephant_ffi.mjs
import * as $regex from "../gleam_stdlib/gleam/regex.mjs";
import {
	makeError,
} from "./gleam.mjs";

const cache = new Map();

export function persist(name, gen) {
	let value = cache.get(name)
	if (value === undefined) {
		value = gen();
		cache.set(name, value);
	}

	return value;
}
